File: src/nfan/fan/sys/LogRecord.cs
//
// Copyright (c) 2006, Brian Frank and Andy Frank
// Licensed under the Academic Free License version 3.0
//
// History:
//   24 Dec 07  Andy Frank  Creation
//

using System.Text;

namespace Fan.Sys
{
  /// <summary>
  /// LogRecord.
  /// </summary>
  public class LogRecord : FanObj
  {

  //////////////////////////////////////////////////////////////////////////
  // Construction
  //////////////////////////////////////////////////////////////////////////

    public static LogRecord make(DateTime time, LogLevel level, Str logName, Str msg) { return make(time, level, logName, msg); }
    public static LogRecord make(DateTime time, LogLevel level, Str logName, Str msg, Err err)
    {
      LogRecord self = new LogRecord();
      make_(self, time, level, logName, msg, err);
      return self;
    }

    public static void make_(LogRecord self, DateTime time, LogLevel level, Str logName, Str msg) { make_(self, time, level, logName, msg, null); }
    public static void make_(LogRecord self, DateTime time, LogLevel level, Str logName, Str msg, Err err)
    {
      self.m_time    = time;
      self.m_level   = level;
      self.m_logName = logName;
      self.m_message = msg;
      self.m_err     = err;
    }

  //////////////////////////////////////////////////////////////////////////
  // Methods
  //////////////////////////////////////////////////////////////////////////

    public override Type type() { return Sys.LogRecordType; }

    public override Str toStr()
    {
      string ts = m_time.toLocale("hh:mm:ss DD-MMM-YY");
      StringBuilder s = new StringBuilder();
      s.Append('[').Append(ts).Append(']')
       .Append(' ').Append('[').Append(m_level).Append(']')
       .Append(' ').Append('[').Append(m_logName).Append(']')
       .Append(' ').Append(m_message);
      return Str.make(s.ToString());
    }

    public void print() { print(Sys.@out()); }
    public void print(OutStream @out)
    {
      @out.printLine(toStr());
      if (m_err != null) m_err.trace(@out, 2);
    }

  //////////////////////////////////////////////////////////////////////////
  // Fields
  //////////////////////////////////////////////////////////////////////////

    public DateTime m_time;
    public LogLevel m_level;
    public Str m_logName;
    public Str m_message;
    public Err m_err;

  }
}

File: src/sys/java/fan/sys/LogRecord.java
//
// Copyright (c) 2006, Brian Frank and Andy Frank
// Licensed under the Academic Free License version 3.0
//
// History:
//   3 Jul 06  Brian Frank  Creation
//
package fan.sys;

/**
 * LogRecord
 */
public class LogRecord
  extends FanObj
{

//////////////////////////////////////////////////////////////////////////
// Construction
//////////////////////////////////////////////////////////////////////////

  public static LogRecord make(DateTime time, LogLevel level, String logName, String msg) { return make(time, level, logName, msg, null); }
  public static LogRecord make(DateTime time, LogLevel level, String logName, String msg, Err err)
  {
    LogRecord self = new LogRecord();
    make$(self, time, level, logName, msg, err);
    return self;
  }

  public static void make$(LogRecord self, DateTime time, LogLevel level, String logName, String msg) { make$(self, time, level, logName, msg, null); }
  public static void make$(LogRecord self, DateTime time, LogLevel level, String logName, String msg, Err err)
  {
    self.time    = time;
    self.level   = level;
    self.logName = logName;
    self.message = msg;
    self.err     = err;
  }

//////////////////////////////////////////////////////////////////////////
// Methods
//////////////////////////////////////////////////////////////////////////

  public Type type() { return Sys.LogRecordType; }

  public String toStr()
  {
    String ts = ((DateTime)time).toLocale("hh:mm:ss DD-MMM-YY");
    StringBuilder s = new StringBuilder();
    s.append('[').append(ts).append(']')
     .append(' ').append('[').append(level).append(']')
     .append(' ').append('[').append(logName).append(']')
     .append(' ').append(message);
    return s.toString();
  }

  public void print() { print(Sys.out()); }
  public void print(OutStream out)
  {
    synchronized (out)
    {
      out.printLine(toStr());
      if (err != null) err.trace(out, 2, true);
    }
  }

//////////////////////////////////////////////////////////////////////////
// Fields
//////////////////////////////////////////////////////////////////////////

  public DateTime time;
  public LogLevel level;
  public String logName;
  public String message;
  public Err err;

}